drivers/video/ov5640.c

```diff
@@ -897,15 +1127,15 @@ static int ov5640_enum_frmival(const struct device *dev, enum video_endpoint_id
 {
```

Comment:
`ov5640_enum_frmival()` is common for both modes, so we cannot return the results of CSI2 for DVP. You need to implement `enum_frmival() ` for DVP by returning the supported framerates for each supported format (I suppose DVP only supports 30 fps for all formats as of now ?).

- The same for `get_frmival()`, it needs to return the current frame rate in DVP modes
- For `set_frmival()`, we can return -ENOTSUP

Reply:
Maybe branching on `ov5640_is_dvp()` like you did elsewhere would work well.

Reply:
@ngphibang and @josuah

As you commented, it looks like that the 30FPs is used as default for DVP. In DVP the framerate depends on pixel clock. Currently, the init_params_dvp configured the `SC_PLL_CTRL2_REG` with 0x64 (PLL multiplier 100) and `SC_PLL_CTRL3_REG` with 0x13 (PLL root divided by 2 and PLL pre-divider 3). I'll try to measure the current FPS.

The functions `set_frmival, get_frmival`, and `enum_frmival` haven't been implemented on `video_stm32_dcmi.c` driver. That is the reason for those functions haven't broken when the DVP is used. [Check here.](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/video/video_stm32_dcmi.c#L375)

So, since those functions are not implemented on DCMI driver, is this a block to accept this PR? Could this be done in another PR? Thanks!

Reply:
Thank you for the update. No, it's not a blocker. But as being said in my previous comment, we don't need to implement those functions but is it possible to just return a (hardcode) 30 fps for `enum_frmival()` and `get_frmival()` and -ENOTSUP for `set_frmival()` ?

Reply:
Well, I've just rethought. Let the frmival callback not implemented is OK as well (especially now we have the new DEVICE_API, it's safe to do so I think). BTW, should the frmival callback in the ov5640 driver be removed as well if it is not tested (due to frmival in dcmi not implemented) ?

Reply:
> Well, I've just rethought. Let the frmival callback not implemented is OK as well (especially now we have the new DEVICE_API, it's safe to do so I think). BTW, should the frmival callback in the ov5640 driver be removed as well if it is not tested (due to frmival in dcmi not implemented) ?

Hi @ngphibang.

Regarding the DCMI, when those functions are calling an error is returned, as expected. Considering those frmival functions are working with CSI2 driver, I am not sure If it is necessary to remove. 

```
[00:00:00.168,000] <err> main: Unable to get video frame interval
[00:00:00.168,000] <err> main: Unable to enumerate video frame interval
[00:00:00.168,000] <err> main: Unable to set video frame interval
```